Rakefile

```diff
@@ -1,72 +1,50 @@
-require "json"
 require "pathname"
 
-CONFIG_FILE  = Pathname.new(__dir__).join(ENV["CNFFILE"] || "config.json")
 BUILD_ROOT   = Pathname.new(ENV["OBJDIR"] || "build")
 INSTALL_ROOT = Pathname.new(ENV["PREFIX"] || "/opt/ood")
+VENDOR_DIR   = Pathname.new(ENV["VENDOR_DIR"] || BUILD_ROOT.join("vendor").expand_path )
 
-def all_components
-  JSON.parse(CONFIG_FILE.read).map { |c| Component.new(c) }
-end
-
-class Component
-  attr_reader :name
-  attr_reader :url
-  attr_reader :tag
-
-  def initialize(opts = {})
-    @name = opts.fetch("name") { raise ArgumentError, "No name specified. Missing argument: name" }.to_s
-    @url  = opts.fetch("url")  { raise ArgumentError, "No url specified. Missing argument: url" }.to_s
-    @tag  = opts.fetch("tag")  { raise ArgumentError, "No tag specified. Missing argument: tag" }.to_s
-    @app  = opts.fetch("app", nil)
-  end
-
-  def app?
-    !!@app
-  end
-
-  def build_root
-    app? ? BUILD_ROOT.join("apps", name) : BUILD_ROOT.join(name)
-  end
-
-  def download_url
-    "#{url}/archive/#{tag}.tar.gz"
-  end
+task :default => :build
 
-  def file_url
-    "apps/#{name}/."
-  end
+def apps
+  %w(activejobs bc_desktop dashboard file-editor files myjobs shell).map {|name| BUILD_ROOT.join("apps", name) }
 end
 
-task :default => :build
+def build_apps
```

Comment:
1. `PASSENGER_APP_ENV=production bin/setup` is only needed, as the default values for `PASSENGER_BASE_URI` for each app are properly set in each app's `bin/setup`, so that will simplify this some
2. This should be turned back into tasks: a task for building all the apps, and a task for building each app. That way the tasks can be run in parallel due to the use of -mj flags when executing the rake task.

Reply:
But the tasks don't have to be "file" tasks. `PASSENGER_APP_ENV=production bin/setup` is safe to re-run and any optimization by using `file` would only be possibly beneficial to people installing only from source.